adapter/src/lib.rs

```diff
@@ -194,7 +195,9 @@ impl EthereumChannel {
         contract_addr: &str,
         balance_root: &str,
     ) -> Result<[u8; 32], Box<dyn Error>> {
-        get_signable_state_root(contract_addr, balance_root)
+        let root = <[u8; 32]>::from_hex(balance_root).unwrap();
```

Comment:
at least `expect`? Although it might be good to handle the errors even if we are sure that `contact_addr` & `balance_root` are ok.